task-01/src/containers/Form.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useReducer, useEffect } from "react";
+
+function reducer(state, action){
+    switch (action.type) {
+      case "fieldChange": {
+        const newState = {...state, [String(action.payload.name)]: String(action.payload.value)}
+        return newState;
+      }
+
+      case "setData": {
+          const newState = action.payload;
+          return newState;
+      }
+      default: {
+        return {...state};
```

Comment:
it's a new object that contains exactly the same data as state itself. `return state` will be an easier option